src/pyvalues.c

```diff
@@ -42,6 +50,26 @@
     PyMem_Free(host);                                                          \
   } while (0)
 
+#define CPY_BUILD_META_FUNC(meta_type, func, val_type)                         \
```

Comment:
This is some serious macro magic. Please move the macro definition and invocation closer together so it's easier to understand what's going on.